pypesto/result.py

```diff
@@ -24,7 +23,7 @@ def __init__(self):
 
     def append(
             self,
-            optimizer_result: 'pypesto.OptimizerResult'
+            optimizer_result,
```

Comment:
shouldn't it be possible to keep these typehints in by loading respective modules in the sphinx conf.py?

Reply:
I will noqa them for the moment. In the long run, maybe these should be part of the result submodule.

Reply:
Doesn't importing `pypesto.optimize`, or whatever module they are available from, in `doc/conf.py`  fix the issue? I thought ForwardReferences would adress exactly this issue.

Reply:
Not sure if I understand. We could import pypesto.xyz in conf.py to enable relative doc links, did not check that. Here in result.py, we cannot import pypesto.optimize.Anything, as this creates circular imports afaik.

Reply:
Now, I added a typehint `optimizer_result: 'pypesto.optimize.OptimizerResult',  # noqa: F821` in here.